### PR TITLE
PP-730: log_err messages do not appear in syslog if PBS_LOCALLOG is n…

### PR DESCRIPTION
**Pull request by prakashcv13**
…ot set and PBS_SYSLOG is set to 1 in pbs.conf

#### Issue-ID

* **[PP-730](https://pbspro.atlassian.net/browse/PP-730)**

#### Problem description
* logic to write logs to the system log is under log_record(). However, log_err() checks if the local log is opened for logging or not. If not, it does not call log_record() and returns.

#### Cause / Analysis
* Analysis is in the problem description above.

#### Solution description
* before returning from log_err(), check if logging to system log is enabled, if yes, record the log.

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [x] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by prakashcv13** [comment]
Yes, you are right. I have done the changes and tested them. Steps to verify the fix are also update in JIRA.

File: src/lib/Liblog/pbs_log.c [hunk 1]
```diff
@@ -772,9 +784,20 @@ log_joberr(int errnum, const char *routine, const char *text, const char *pjid)
...
 		(void)log_open("/dev/console", log_directory);
 
-		if (log_opened < 1)
+		if (log_opened < 1) {
+#if SYSLOG
+			char slogbuf[LOG_BUF_SIZE] = {0};
+			if (syslogopen != 0) {
+				snprintf(slogbuf, LOG_BUF_SIZE,
+					"%s;%s;%s\n",
+					class_names[PBS_EVENTCLASS_SERVER],
```
Comment (arungrover) [inline src/lib/Liblog/pbs_log.c]: This needs to be EVENTCLASS_JOB not server. Rest looks good

Comment (prakashcv13) [inline src/lib/Liblog/pbs_log.c]: copy paste :). Corrected in the latest commit. Updated test logs attached to the ticket as well.

Thanks!

**Comment by prakashcv13** [comment]
updated the version in the ticket. Will wait for a second review. Thanks!

**Comment by prakashcv13** [comment]
Hi All,

I need one more reviewer for this simple fix. Any takers?

Thanks,
Prakash

**Comment by prakashcv13** [comment]
@PBSPro/pbspro-maintainers, I have squashed my commits and rebased my branch, could you please take it forward.

**Comment by prakashcv13** [comment]
@mike0042 - Thank you!
@subhasisb, there are two license/cla checks - one is shown as successfule and 1 as expected. Could you please help me understand how we fix this?

Thanks,
Prakash

**Comment by prakashcv13** [comment]
@PBSPro/pbspro-maintainers,

I have two approvals and the branch is rebased with the master. Could you please merge this PR.

Thanks,
Prakash

**Comment by subhasisb** [comment]
Merged!

**Comment by prakashcv13** [comment]
Thank you @subhasisb.

**Pull request merged**